### PR TITLE
feat: [macOS] ComboBox

### LINKED ISSUE #2972: [macOS] ComboBox
## What would you like to be added:

ComboBox support.

## Why is this needed:

Feature parity.

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #2972 

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

ComboBox shows up, but does not display items.

## What is the new behavior?

ComboBox works properly.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.macOS.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	public partial class ItemsControl : Control
+	{
+		partial void InitializePartial()
+		{
+
+		}
+
+		partial void RequestLayoutPartial()
+		{
+			NeedsLayout = true;
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.macOS.cs:23]: Use `InvalidateMeasure()` instead, `NeedsLayout` does not propagate the invalidation to the parents.

File: src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.macOS.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+			current.RemoveFromSuperview();
+		}
+
+		public override void Layout()
+		{
+			UpdateItemsIfNeeded();
+			base.Layout();
+		}
+
+		public override CGSize SizeThatFits(CGSize size)
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.macOS.cs:37]: `SizeThatFits` may not be called properly on macOS, it's better to override `OnMeasure` instead.

File: src/Uno.UI/UI/Xaml/FrameworkElement.macOS.cs [hunk 3]
```diff
@@ -155,7 +155,7 @@ protected virtual void OnAfterArrange()
 			return result.LogicalToPhysicalPixels();
 		}
 
-		public CGSize SizeThatFits(CGSize size)
+		public virtual CGSize SizeThatFits(CGSize size)
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/FrameworkElement.macOS.cs:158]: This is most likely not needed as per the comment above.

**Comment by jeromelaban** [comment]
Thanks Martin!

**Comment by jeromelaban** [comment]
Seems like the build failed because of the change in SizeThatFits.

File: src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.macOS.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.macOS.cs:23]: Fixed

File: src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.macOS.cs (hunk 2 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.macOS.cs:37]: Fixed

File: src/Uno.UI/UI/Xaml/FrameworkElement.macOS.cs (hunk 3 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/FrameworkElement.macOS.cs:158]: And fixed :-)

**Pull request merged**